src/libraries/System.Private.Xml/src/System/Xml/Serialization/ImportContext.cs

```diff
@@ -156,7 +156,7 @@ internal StringCollection Warnings
                 else
                 {
                     Warnings.Add(SR.Format(SR.XmlMismatchSchemaObjects, item.GetType().Name, qname.Name, qname.Namespace));
-                    Warnings.Add("DEBUG:Cached item key:\r\n" + (string?)looks[cachedItem] + "\r\nnew item key:\r\n" + (string?)looks[item]);
+                    Warnings.Add($"DEBUG:Cached item key:\r\n{(string?)looks[cachedItem]}\r\nnew item key:\r\n{(string?)looks[item]}");
```

Comment:
Roslyn converts four arguments to string[] and then calls string.Concat with interpolation. While with `+` operator, it emits a single string.Concat call: [sharplab with nightly roslyn build](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAICYCMBYAKAwGYACbUgYUIG9DSHyyMcAGUgWRwAoX2APAJSkAvAD5SAIgQB7ANYBTGCCmkA1KX7rN2/gG56jEuTacsvU0NESAJNPlKVNfgF9Sztx8kGCLoA===).

String interpolation is not necessarily the best alternative in all cases.

Reply:
> Roslyn converts four arguments to string[] and then calls string.Concat with interpolation. While with + operator, it emits a single string.Concat call: sharplab with nightly roslyn build.

That's a nightly build of the compiler but targeting .NET 5.  You'd see different results if you were targeting .NET 6.
https://devblogs.microsoft.com/dotnet/string-interpolation-in-c-10-and-net-6/